nuxeo-java-client/src/main/java/org/nuxeo/client/util/Dates.java

```diff
@@ -120,79 +108,52 @@ public static String formatW3CDateTimeMs(Date date) {
         }
         Calendar cal = new GregorianCalendar(TimeZone.getTimeZone("UTC"));
         cal.setTime(date);
-        StringBuilder buf = new StringBuilder(32);
-        return buf.append(cal.get(Calendar.YEAR))
-                  .append('-')
-                  .append(pad(cal.get(Calendar.MONTH) + 1))
-                  .append('-')
-                  .append(pad(cal.get(Calendar.DATE)))
-                  .append('T')
-                  .append(pad(cal.get(Calendar.HOUR_OF_DAY)))
-                  .append(':')
-                  .append(pad(cal.get(Calendar.MINUTE)))
-                  .append(':')
-                  .append(pad(cal.get(Calendar.SECOND)))
-                  .append('.')
-                  .append(pad3(cal.get(Calendar.MILLISECOND)))
-                  .append('Z')
-                  .toString();
+        return String.format("%04d-%02d-%02dT%02d:%02d:%02d:%03dZ", cal.get(Calendar.YEAR), cal.get(Calendar.MONTH) + 1,
+                cal.get(Calendar.DATE), cal.get(Calendar.HOUR_OF_DAY), cal.get(Calendar.MINUTE),
+                cal.get(Calendar.SECOND), cal.get(Calendar.MILLISECOND));
     }
 
-    private final static String pad(int i) {
-        return i < 10 ? "0".concat(String.valueOf(i)) : String.valueOf(i);
-    }
-
-    private final static String pad3(int i) {
-        if (i < 10) {
-            return "00".concat(String.valueOf(i));
-        } else if (i < 100) {
-            return "0".concat(String.valueOf(i));
-        } else {
-            return String.valueOf(i);
-        }
-    }
-
-    private final static int readYear(Calendar cal, String str, int off) throws ParseException {
+    private static int readYear(Calendar cal, String str, int off) throws ParseException {
```

Comment:
`private` -> `protected` for all methods. But I don't understand why we have all this low-level parsing code when even `SimpleDateFormat` can do it.